environ/environ.go

```diff
@@ -82,3 +116,107 @@ func (e *Environ) Load(s store.Store, service string, collisions *[]string) erro
 func (e *Environ) LoadNoPaths(s store.Store, service string, collisions *[]string) error {
 	return e.load(s, service, collisions, true)
 }
+
+// LoadStrict loads all services from s in strict mode: env vars in e with value equal to valueExpected
+// are the only ones substituted. If there are any env vars in s that are also in e, but don't have their value
+// set to valueExpected, this is an error.
+func (e *Environ) LoadStrict(s store.Store, valueExpected string, pristine bool, services ...string) error {
+	return e.loadStrict(s, valueExpected, pristine, false, services...)
+}
+
+// LoadNoPathsStrict is identical to LoadStrict, but uses v1-style "."-separated paths
+//
+// Deprecated like all noPaths functionality
+func (e *Environ) LoadStrictNoPaths(s store.Store, valueExpected string, pristine bool, services ...string) error {
+	return e.loadStrict(s, valueExpected, pristine, true, services...)
+}
+
+func (e *Environ) loadStrict(s store.Store, valueExpected string, pristine bool, noPaths bool, services ...string) error {
+	for _, service := range services {
+		rawSecrets, err := s.ListRaw(strings.ToLower(service))
+		if err != nil {
+			return err
+		}
+		err = e.loadStrictOne(rawSecrets, valueExpected, pristine, noPaths)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (e *Environ) loadStrictOne(rawSecrets []store.RawSecret, valueExpected string, pristine bool, noPaths bool) error {
+	parentMap := e.Map()
+	parentExpects := map[string]struct{}{}
+	for k, v := range parentMap {
+		if v == valueExpected {
+			if k != normalizeEnvVarName(k) {
+				return ErrExpectedKeyUnnormalized{Key: k, ValueExpected: valueExpected}
+			}
+			// TODO: what if this key isn't chamber-compatible but could collide? MY_cool_var vs my-cool-var
+			parentExpects[k] = struct{}{}
+		}
+	}
+
+	envVarKeysAdded := map[string]struct{}{}
+	for _, rawSecret := range rawSecrets {
+		envVarKey := secretKeyToEnvVarName(rawSecret.Key, noPaths)
+
+		parentVal, parentOk := parentMap[envVarKey]
+		// skip injecting secrets that are not present in the parent
+		if !parentOk {
+			continue
```

Comment:
Could we log this with verbose as well?

Reply:
Maybe, but that would mean getting `verbose` into this lib code somehow, or a logger. Not sure that's worth doing?

Reply:
Ah yeah that doesn't sound worth it 👍

Reply:
From `exec`'s point of view, I'm pretty sure this would not be surprising. If the user's got a secret in the parent that we're dropping, either:

1. User didn't want it and everything is as it should be.
2. User did want it but forgot to write a `chamberme`. Hopefully the app crashes when it can't find its DB password.

If #2 happens undetected, well, that's 2 checks chamber did, but at least it didn't inject tokens unasked for.